src/dmd/globals.d

```diff
@@ -469,6 +469,24 @@ nothrow:
                FileName.equals(filename, loc.filename);
     }
 
+    extern (D) bool opEquals(Loc loc) const pure
+    {
+        return charnum == loc.charnum &&
+               linnum == loc.linnum &&
+               FileName.equals(filename, loc.filename);
+    }
+
+    extern (D) hash_t toHash() const pure
+    {
+        import dmd.root.hash;
+        import dmd.utils;
+
+        auto hash = hashOf(linnum);
+        hash = mixHash(hash, hashOf(charnum));
+        hash = mixHash(hash, hashOf(filename.toDString));
```

Comment:
Please tell me there's something like mixHash() in druntime.

Reply:
druntime's explicit `mixHash` function [was removed October 14, 2014](https://github.com/dlang/druntime/commit/6da8b3b367531670bcc1e8d1ad3579737cfef7bd#diff-1cb07fafd18644ca9974aea7a74483e9) but druntime `hashOf` has a two-argument form that can be used for this:
```d
hash = hashOf(charnum, hash);
```